3-UsingAPIs/Week1/homework/ex5-pokerDiceChain.js

```diff
@@ -2,22 +2,22 @@ const rollDie = require('../../helpers/pokerDiceRoller');
 
 function rollDice() {
   const results = [];
+  const successHandler = (value, promise, die) => {
+    results.push(value);
+    return promise(die);
+  };
```

Comment:
Yes, this is the function I had in mind. But this `successHandler()` function (excellent name), does not really require the second parameter `promise`, does it? Or did you perhaps anticipate that you would potentially want to pass something else than `rollDie`? If not, then this would work fine too and be more succinct.

```suggestion
  const successHandler = (value, die) => {
    results.push(value);
    return rollDie(die);
  };
```

For readability, it would be nice to add a blank line just before and after the handler function, to make it stand out from the rest of the code inside `rollDice()`.

Reply:
Thank you so much for your feedback Jim. I really appreciate your help.